### PR TITLE
Add giveawayDeleted event and edit embed when rerolling

### PR DESCRIPTION
**Pull request by ShreyashKira**
* Added giveawayDeleted event (use case: some extra data we saved that deleted after giveaway ends left if giveaway is deleted)
* Added extra data column name as requirements so that if someone want to save some extra data like bot id or anything then they have no boundaries
* Edit embed after every reroll so that we can use the data perfectly.
* And nothing else :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,6 +107,7 @@ client.on("message", (message) => {
 * **options.embedColor**: a hexadecimal color for the embeds of giveaways.  
 * **options.embedColorEnd**: a hexadecimal color the embeds of giveaways when they are ended.  
 * **options.reaction**: the reaction that users will have to react to in order to participate.  
+* **options.requirements**: Extra data which you want to save regarding this giveaway. You can access it from the giveaway object of the giveaway created by giveaway.requirements.
```
Comment (Androz2091) [inline README.md:110]:
```suggestion
* **options.extraData**: Extra data which you want to save regarding this giveaway. You can access it from the giveaway object using `giveaway.extraData`.
```

File: src/Giveaway.js [hunk 2]
```diff
@@ -78,6 +78,11 @@ class Giveaway extends EventEmitter {
          * @type {GiveawayMessages}
          */
         this.messages = options.messages;
+        /**
+         * Requirements section for extra data
```
Comment (Androz2091) [inline src/Giveaway.js:82]:
```suggestion
         * Extra data concerning this giveaway
```

File: src/Giveaway.js [hunk 3]
```diff
@@ -78,6 +78,11 @@ class Giveaway extends EventEmitter {
...
+         * @type {Requirements}
```
Comment (Androz2091) [inline src/Giveaway.js:83]:
```suggestion
         * @type {any}
```

File: src/Giveaway.js [hunk 4]
```diff
@@ -78,6 +78,11 @@ class Giveaway extends EventEmitter {
...
+         */
+        this.requirements = options.requirements;
```
Comment (Androz2091) [inline src/Giveaway.js:85]:
```suggestion
        this.extraData = options.extraData;
```

File: src/Manager.js [hunk 5]
```diff
@@ -159,7 +159,8 @@ class GiveawaysManager extends EventEmitter {
                 exemptPermissions: options.exemptPermissions,
                 exemptMembers: options.exemptMembers,
                 embedColor: options.embedColor,
-                embedColorEnd: options.embedColorEnd
+                embedColorEnd: options.embedColorEnd,
+                requirements: options.requirements
```
Comment (Androz2091) [inline src/Manager.js:163]:
```suggestion
                extraData: options.extraData
```

File: src/Manager.js [hunk 6]
```diff
@@ -252,6 +253,7 @@ class GiveawaysManager extends EventEmitter {
             }
             this.giveaways = this.giveaways.filter((g) => g.messageID !== messageID);
             await this.deleteGiveaway(messageID);
+            this.emit('giveawayDeleted', giveaway)
```
Comment (Androz2091) [inline src/Manager.js:256]:
```suggestion
            this.emit('giveawayDeleted', giveaway);
```

**Comment by ShreyashKira** [comment]
😅 Actually the time when I was coding this I planned to use extraData for requirements but after some days I realize that it can be extended to provide more powers.

**Pull request merged**